src/measure_theory/outer_measure.lean

```diff
@@ -357,6 +484,91 @@ theorem le_of_function {μ : outer_measure α} :
   le_trans (μ.mono hs) $ le_trans (μ.Union f) $
   ennreal.tsum_le_tsum $ λ i, H _⟩
 
+lemma is_greatest_of_function :
+  is_greatest {μ : outer_measure α | ∀ s, μ s ≤ m s} (outer_measure.of_function m m_empty) :=
+⟨λ s, of_function_le _, λ μ, le_of_function.2⟩
+
+lemma of_function_eq_Sup : outer_measure.of_function m m_empty = Sup {μ | ∀ s, μ s ≤ m s} :=
+(@is_greatest_of_function α m m_empty).is_lub.Sup_eq.symm
+
+/-- If `m u = ∞` for any set `u` that has nonempty intersection both with `s` and `t`, then
+`μ (s ∪ t) = μ s + μ t`, where `μ = measure_theory.outer_measure.of_function m m_empty`. -/
+lemma of_function_union_of_separated {s t : set α}
```

Comment:
I don't really understand the mathematical content of this lemma. Its assumptions are satisfied when any nonempty subset of `s` has infinite measure, and any nonempty subset of `t` has infinite measure, right? So I don't see in which interesting case it can be applied (right now, there is no other example of use of this lemma in the PR), and I don't see why there is `separated` in the name (it would for instance apply when `s = t = univ` on an infinite-measure space where the sigma-algebra is the trivial one, made of `univ` and the empty set, in which case there is no separation).

Reply:
The use case I have in mind (and in a local branch) is `m = ∞` on sets of `diam > ε` while `ε < ⨅ (x ∈ s) (y ∈ t), edist x y`.  What name would you prefer?

Reply:
So for instance your `m` would be infinite on a two points set `{a, b}` with `a` in `s` and `b` in `t`? This seems unusual enough that the docstring deserves to be expanded , explaining your example use case with more details. As for the name, something like `of_function_union_of_top_of_nonempty_inter` would maybe be better.

Reply:
Done.